common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2922,6 +2920,10 @@ def _copy_subdag(self, user_id, destination_version, block_key, source_blocks, d
                 raw=True,
                 block_defaults=new_block.defaults
             )
+            # _new_block doesn't copy the original_usage data needed for library blocks:
+            if new_block.edit_info.original_usage:
+                destination_block.edit_info.original_usage = new_block.edit_info.original_usage
+                destination_block.edit_info.original_usage_version = new_block.edit_info.original_usage_version
```

Comment:
@bradenmacdonald Thanks for the explanations. Seems fair to create a separate ticket for the issue with duplication - can you add it to the Jira backlog? It's also probably worth pinging Marco on the bug to make him aware, and make sure course authors using the content libraries are aware of the limitation in the meantime. That makes me :+1: once the comments around `_new_block` are addressed.

Do we know why the CMS recreates the item itself, instead of delegating the copy operation to the modulestore? For `don't_duplicate_fields`, another more generic option would be to allow an xblock to override the default duplication mechanism, a bit like how XML serialization is handled.